### PR TITLE
node-fs-readfile - Roland Estropia Jr.

### PR DESCRIPTION
**Pull request by restropiajr**
image

### CODE REVIEW

File: node-fs-readfile/cat.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { readFile } from 'node:fs/promises';
+
+async function readCatAnyFiles() {
+  const fileArray = process.argv.slice(2);
+
+  const promiseArray = fileArray.map(async (file) => {
+    const filePath = new URL(file, import.meta.url);
+    const contents = await readFile(filePath, { encoding: 'utf8' });
+    return contents;
```
Comment (RobertGardner) [inline node-fs-readfile/cat.js]: async/await here works, but it's not necessary. `readFile` returns a Promise, which is what you want to pass to Promise.all.

Also, although the Node documentation may look like you need import.meta.url, etc., it actually isn't necessary. By default, `readFile` will look in the current directory for the file.

So you can just use `fileArray.map((file) => readFile(file, 'utf8'))`.

(Also note that as a shortcut, `readFile` allows you to send the encoding as a string in the second parameter.)

File: node-fs-readfile/cat.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { readFile } from 'node:fs/promises';
+
+async function readCatAnyFiles() {
+  const fileArray = process.argv.slice(2);
```
Comment (RobertGardner) [inline node-fs-readfile/cat.js]: It's best not to put the data type in the variable name. The reader can usually easily figure out what data type the variable holds, so putting it in the name makes the code more verbose and actually harder to read. Just `files` would be a great name.

File: node-fs-readfile/cat.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const fileContents = await Promise.all(promiseArray);
+  fileContents.forEach((fileContent) => {
+    console.log(fileContent);
+  });
+}
+
+try {
+  await readCatAnyFiles();
+} catch (error) {
+  console.error('Error:', error.message);
```
Comment (RobertGardner) [inline node-fs-readfile/cat.js]: You usually want to log the entire error, not just the message field. That way you will get a stack trace that will help you figure out where things went wrong.

**Review by RobertGardner — APPROVED** [review]

File: node-fs-readfile/cat.js (hunk 1 above)
Comment (restropiajr) [reply node-fs-readfile/cat.js]: Updated

File: node-fs-readfile/cat.js (hunk 2 above)
Comment (restropiajr) [reply node-fs-readfile/cat.js]: Updated

File: node-fs-readfile/cat.js (hunk 3 above)
Comment (restropiajr) [reply node-fs-readfile/cat.js]: Updated

**Pull request merged**